middleware/basic_auth_test.go

```diff
@@ -56,3 +56,16 @@ func TestBasicAuthWrap(t *testing.T) {
 		}
 	})
 }
+
+func TestBasicAuthURLEncoded(t *testing.T) {
```

Comment:
I do not like that you put things always at the end I causes more conflicts than needed and introduces balagan.
This should live below TestBasicAuth or better be a subtest.

Reply:
Well, if you think about, it lived below TestBasicAuth, just not directly... It's a subtest now :)